ARGO/src/AssetManager.cpp

```diff
@@ -0,0 +1,237 @@
+#include "stdafx.h"
+#include "AssetManager.h"
+
+AssetManager* AssetManager::s_instance = NULL;
+
+AssetManager* AssetManager::Instance(SDL_Renderer& t_renderer)
+{
+	if (s_instance == NULL)
+	{
+		s_instance = new AssetManager(t_renderer);
+	}
+	return s_instance;
+}
+
+AssetManager* AssetManager::Instance()
+{
+	return s_instance;
+}
+
+void AssetManager::Release()
+{
+	delete s_instance;
+	s_instance = NULL;
+}
+
+SDL_Texture* AssetManager::GetTexture(std::string t_filename)
+{
+	std::string fullPath = SDL_GetBasePath();
+	fullPath.append(Utilities::IMAGES_PATH + t_filename);
+
+	if (!m_textures[fullPath])
+	{
+		m_textures.at(fullPath) = loadTexture(fullPath);
+	}
+
+	return m_textures[fullPath];
+}
+
+Mix_Chunk* AssetManager::GetSfx(std::string t_filename)
+{
+	std::string fullPath = SDL_GetBasePath();
+	fullPath.append(Utilities::SOUNDS_PATH + t_filename);
+
+	if (!m_sfx[fullPath])
+	{
+		m_sfx.at(fullPath) = loadSfx(fullPath);
+	}
+
+	return m_sfx[fullPath];
+}
+
+Mix_Music* AssetManager::GetMusic(std::string t_filename)
+{
+	std::string fullPath = SDL_GetBasePath();
+	fullPath.append(Utilities::SOUNDS_PATH + t_filename);
+
+	if (!m_music[fullPath])
+	{
+		m_music.at(fullPath) = loadMusic(fullPath);
+	}
+
+	return m_music[fullPath];
+}
+
+SDL_Texture* AssetManager::GetText(std::string t_text, std::string t_filename, int t_size, SDL_Color t_colour)
+{
+	TTF_Font* font = GetFont(t_filename, t_size);
+
+	std::string key = t_text + t_filename + (char)t_size + (char)t_colour.r + (char)t_colour.g + (char)t_colour.b + (char)t_colour.a;
+
+	if (!m_text[key])
+	{
+		m_text[key] = CreateTextTexture(font, t_text, t_colour);
+	}
+	return m_text[key];
+}
+
+TTF_Font* AssetManager::GetFont(std::string t_filename, int t_fontSize)
+{
+	std::string fullPath = SDL_GetBasePath();
+	fullPath.append(Utilities::FONTS_PATH + t_filename);
+	std::string key = fullPath + (char)t_fontSize;
+
+	if (!m_fonts[key])
+	{
+		m_fonts[key] = loadFont(fullPath.c_str(), t_fontSize);
+	}
+
+	return m_fonts[key];
+}
+
+SDL_Texture* AssetManager::CreateTextTexture(TTF_Font* t_font, std::string t_text, SDL_Color t_colour)
+{
+	SDL_Surface* surface = TTF_RenderText_Solid(t_font, t_text.c_str(), t_colour);
+
+	if (surface == NULL)
+	{
+		printf("Text Render Error: %s\n", TTF_GetError());
+		return NULL;
+	}
+
+	SDL_Texture* tex = SDL_CreateTextureFromSurface(&m_renderer, surface);
```

Comment:
tex is maybe a bit ambigious as you're working with texts and textures

Reply:
same in load texture function

Reply:
What? they're both textures but m_text are textures for text objects while m_textures are textures for images.
"tex" is a shortened name for SDL_Texture...